File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cs
public class Solution {
    public int FindMin(int[] nums) {
        if(nums.Length==1)
        return nums[0];
        int left=0,right=nums.Length-2;
        while(left<=right)
        {
            int mid=left+(right-left)/2;
            if(nums[mid]>=nums[0])
            left=mid+1;
            else 
            right=mid-1;
        }
        return nums[left]<nums[0]?nums[left]:nums[0];
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0, right = nums.size() - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            nums[mid] < nums[0] ? right = mid - 1 : left = mid + 1;
        }

        return left == nums.size() ? nums[0] : nums[left];
    }
};